make-rules/shared-macros.mk

```diff
@@ -1244,6 +1244,24 @@ CFLAGS +=	$(CC_BITS)
 # Add compiler specific 'default' features
 CFLAGS +=	$(CFLAGS.$(COMPILER))
 
+# Enable CTF if desired
+ifeq ($(strip $(USE_CTF)),yes)
+# -gdwarf-4 is the newest DWARF revision understood by illumos libctf
+#
+# -gstrict-dwarf prevents the compiler from including newer DWARF features
+#
+# -msave-args makes it possible for debugging tools (specifically pstack and
+# mdb) to retrieve function arguments, and is required for the python pstack
+# hooks.
+CFLAGS += -g -gdwarf-4 -gstrict-dwarf -msave-args
+# By default, GCC may put parts of functions in different named sub-sections
+# of .text based on their presumed or measured calling frequency.  This is not
+# currently handled by libctf and other illumos tools which assume that a
+# function's text lies in a single contiguous range of virtual addresses.
+# Disable this optimization if building with CTF
+CFLAGS += -fno-reorder-functions -fno-reorder-blocks-and-partition
```

Comment:
Are these supported by `clang`?

Reply:
Not listed in their documentation (looked at https://clang.llvm.org/docs/ClangCommandLineReference.html)

clang appears to have a roughly equivalent -f{no-,}split-machine-functions, documented as 
"Enable late function splitting using profile information (x86 ELF)".